packages/core/src/machines/index.ts

```diff
@@ -772,7 +804,31 @@ export const createAuthMachine = ({
             })
           }
         },
-
+        signUpSecurityKey: async (_, { email, options }) => {
+          if (!isValidEmail(email)) {
+            return Promise.reject<SignUpResponse>({ error: INVALID_EMAIL_ERROR })
+          }
+          // TODO anonymous users
+          const nickname = options?.nickname
+          if (nickname) delete options.nickname
```

Comment:
Could you please elaborate on why deleting `options.nickname` is necessary here?

Reply:
the `'/signup/webauthn` accepts any property in `SignUpOption`, meaning `nickname` is not allowed. I set a lax validation in hasura-auth, meaning unknown properties will be ignored rather than raise an error, but it's better not to send it.
I added an explanation in the code

Reply:
Thanks for the explanation. This however raises another question: Why do we even need this to be part of `SignUpSecurityKeyOptions`? I just checked the code and don't see this being used anywhere. It's just simply removed here and that's it. Maybe I'm missing something here.

Reply:
The option was available in the react hook but not correctly typed in `nhost.auth.signUp`.
I just pushed a fix.
You can now write:
```js
await nhost.auth.signUp({ email: 'bob@example.com', securityKey: true, options: { nickname: 'my laptop' } })
```